### ISSUE TITLE
Question about GRPO training approach and code availability

### ISSUE THREAD

**Issue by buchuoer** [zli12321/VideoHallu#1]
Hi, I'm wondering whether you directly perform GRPO training on the mixed dataset using Qwen-2.5-VL-7B, or if you first conduct pre-training (similar to Video-R1) before applying GRPO? By the way, are you planning to open-source the code? Thanks！

**Comment by zli12321** [comment]
We directly perform GRPO training on the mixed data without supervised fine-tuning. The code is now uploaded to the repo, it's similar to the video-R1 code, with changes of reward function and dataset path.

**Comment by buchuoer** [comment]
Thank you. However, it seems that the paper mentions a three-stage training process for the model, but looking at the code, there doesn't appear to be a stage that uses Video-LLaVA data for training. Did I miss something?

**Comment by buchuoer** [comment]
What are the differences between the roles of Categorization-1 and Categorization-2 here? Why is Categorization-1 a coarse-grained classification while Categorization-2 is fine-grained? So, which category should be followed during evaluation—should we only use Categorization-2？

{"RowID": "59", "Model": "Runawaygen2", "Caption Notes": "", "annotator": "wxy", "video_path": "./Runawaygen2/1uPVDzHcNEAApEhzlewBnx639vlNPHhLL.mp4", "Prompt": "Generate a video for launching two rockets", "Question": "How many rockets are launched in the video?", "Answer": "Two", "Categorization-1": "Alignment", "Categorization-2": "Alignment - Entity Counting"}
{"RowID": "59", "Model": "Runawaygen2", "Caption Notes": "", "annotator": "wxy", "video_path": "./Runawaygen2/1uPVDzHcNEAApEhzlewBnx639vlNPHhLL.mp4", "Prompt": "Generate a video for launching two rockets", "Question": "Do all rockets launch at the same pace and clear the tower simultaneously?", "Answer": "No, one of them is left behind and still held by the launch clamp.", "Categorization-1": "Physics", "Categorization-2": "Spatial-temporal Consistency - Temporal Dynamics"}
{"RowID": "59", "Model": "Runawaygen2", "Caption Notes": "", "annotator": "wxy", "video_path": "./Runawaygen2/1uPVDzHcNEAApEhzlewBnx639vlNPHhLL.mp4", "Prompt": "Generate a video for launching two rockets", "Question": "Are there any fast-moving objects in this video?", "Answer": "Yes, there are two launching rockets.", "Categorization-1": "Alignment", "Categorization-2": "Spatial-temporal Consistency - Spatial Dynamics"}

**Comment by zli12321** [comment]
Categorization-1 is the general categorization. Categorization-2 is the more fine-grained categorization. In our paper, we show the accuracy for both categorizations. What categorization to use depends on your specific context. In general, reporting the accuracy for categorization-1 is sufficient

**Comment by zli12321** [comment]
(Table 3 Page 8) In the paper, for the training, there are only one training stage for GRW-R1, PRW-R1, and SR-R1. For these three models, we train them using GRPO on the corresponding datasets. 

For (GRW+SR-R1) and (PRW+SR), we first train the model on the first data using GRPO, save the weights, then load the weights and train it on the second dataset.

**Comment by buchuoer** [comment]
Thank you, I understand. Regarding the categorization, I still have a question: Why are the former and latter inconsistent? Shouldn't they belong under the same category? For example:
"Categorization-1": "Alignment",
"Categorization-2": "Spatial-temporal Consistency - Spatial Dynamics"

**Comment by wuxiyang1996** [comment]
We perform data labeling using GPT and in-context learning, using our pre-defined categorization given in our paper. Such a pipeline may result in some inconsistencies between categorization 1 and 2. Please refer to the results in Categorization-2.

**Comment by buchuoer** [comment]
Sorry, I have quite a few questions. How do you conduct your evaluation? In footnote 2 of the paper, it states, "We then do a train/test split on our data with 800 train QA pairs and 1,525 test pairs. We ensure that our train/test split do not have video overlaps." This only involves 2,325 pairs, but you have a total of 3,233 question-answer pairs. Are the final test results for multiple models conducted on these 1,525 pairs? Additionally, could you share your test code? Thank you.

**Comment by buchuoer** [comment]
Oh, I noticed that the set_split field in VideoHallu/Data/data.jsonl records the data splits. Is the data corresponding to the dev value completely unused?

**Comment by zli12321** [comment]
The dev data is validation during training. We mostly use the test data to evaluate models.

**Comment by buchuoer** [comment]
“Use our formatted JSON files (synthetic_data_split.json and physbench_train_split.json) and follow their setup to train a model.
Run training script”
Where is the validation set used in this code? I can’t find it.More details, please.

**Comment by buchuoer** [comment]
could you share your test code? Thank you.

**Comment by zli12321** [comment]
https://github.com/zli12321/videoHallu_inference/tree/main

It's pretty naive and simple for loop. I'd recommend checking out individual models' website to set up environment.

**Comment by buchuoer** [comment]
The synthetic_data_split.json only contains 800 pieces of data marked as "train." This is the only data passed during the synthetic data training phase. Does this mean the "dev" marked data is not used?

**Comment by zli12321** [comment]
Dev is not used for training.

**Comment by buchuoer** [comment]
Can you open-source your model?

**Comment by wuxiyang1996** [comment]
We are working on retraining the model, and we will release the model soon. The previous model we trained was deleted by the cloud platform.

**Comment by buchuoer** [comment]
Could you share training visualizations (e.g., loss, accuracy trends)?

**Comment by wuxiyang1996** [comment]
We are under constraints regarding releasing more details about our paper. Please stay tuned for our further updates. We will release all the necessary training details and models once the constraints are released.

**Comment by buchuoer** [comment]
This is my training curve of SR. Is this normal? The accuracy doesn't seem to have improved significantly. The accuracy of the answers after reflection has decreased compared to the direct answers from qwen7b. Under direct answering, it's about the same as the base model.
<img width="1080" height="600" alt="Image" src="https://github.com/user-attachments/assets/105b4963-7e70-4100-95d9-b27debe85b3f" />

**Comment by buchuoer** [comment]
Looking at your code, it seems you've attempted to perform SFT directly on SR. What were your results at that time?
